cpp/src/arrow/dataset/rados.cc

```diff
@@ -0,0 +1,75 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
This file (and its header) should maybe be in `src/arrow/util` or `src/arrow/filesystem`.

Reply:
Maybe we can put this in an internal file like `file_skyhook_internal.h` which would contain this along with the RadosConnect interface ?

Reply:
That could work although it wouldn't be my first conclusion.  Elsewhere the file_xyz files are about bridging the gap between a specific format (which has readers and writers in an `arrow/parquet` or `arrow/csv` or `arrow/ipc` directory and these formats know nothing of datasets) and the generic datasets API (e.g. file_base, dataset.h, etc.  These common files know nothing about any specific format).

So my thinking was more that you might not have enough in just one file to create an `arrow/rados` directory but the format-specific files should go somewhere other than `arrow/dataset`.  At a quick inventory we have:

Types that help when interacting with rados as a client (in rados.cpp/rados.h, these are analogous to our hdfs or s3 client libraries which live in `src/arrow/filesystem`.  I think `RadosConnection` and `SkyhookDirectObjectAccess` belong here as well (this might have been your point).  These types can stand on their own as Ceph integration types even if skyhook doesn't exist.

Types that define the "skyhook protocol" (the flatbuffer files, the global/static methods `SerializeScanRequest`, `DeserializeScanRequest`, `SerializeTable`, and `DeserializeTable`).  Personally, I think these might be best to live in the same directory as the cls.

Types that bridge the dataset API and the skyhook API (this would be `SkyhookFileFormat` and it uses types from both of the above categories).